govcd/filter_helpers.go

```diff
@@ -496,7 +496,7 @@ func strToRegex(s string) string {
 // If the value looks like a number, or a true/false value, the corresponding type is returned
 // Otherwise, we assume it's a string.
 // We do this because the API doesn't return the metadata type
```

Comment:
I just read this. Could it be due to the wrong mapping of XsiType that I changed in this PR?